root/programs/C#/Frameworks/Infrastructure/ServiceInterface/AsyncProcessingService/AsyncProcessingService.cs

```diff
@@ -99,126 +109,223 @@ private void InitializeComponent()
-                    _asyncParameterValue.ProgressRate = 0;
-                    _asyncParameterValue.StatusId = (int)AsyncProcessingServiceParameterValue.AsyncStatus.AbnormalEnd;
-                    _asyncParameterValue.CommandId = 0;
-                    break;
-                case AsyncProcessingServiceParameterValue.AsyncStatus.Abort:
-                    _asyncParameterValue.ProgressRate = 0;
-                    _asyncParameterValue.StatusId = (int)AsyncProcessingServiceParameterValue.AsyncStatus.Abort;
-                    _asyncParameterValue.CommandId = 0;
-                    break;
-            }
-            _asyncParameterValue.NumberOfRetries = numberOfRetries;
-            _asyncParameterValue.CompletionDateTime = completionDateTime;
+        #endregion
 
-            return _asyncParameterValue;
-        }
+        #region MainThreadInvoke
 
         /// <summary>
-        /// Set things in motion so your service can do its work.
+        /// calling main thread
         /// </summary>
-        protected override void OnStart(string[] args)
+        public void MainThreadInvoke()
         {
-            StartByThreadPool(AsyncProcessingServiceParameterValue.AsyncStatus.Processing);
+            _asyncParameterValue = new AsyncProcessingServiceParameterValue("AsyncProcessingService", "Select", "Select", "SQL",
+                                                                            new MyUserInfo("AsyncProcessingService", "AsyncProcessingService"));
+
+            AsyncProcessingService.LayerB myBusiness = new AsyncProcessingService.LayerB();
+            AsyncProcessingServiceReturnValue asyncReturnValue = (AsyncProcessingServiceReturnValue)myBusiness.DoBusinessLogic(
+                                                                 (BaseParameterValue)_asyncParameterValue, DbEnum.IsolationLevelEnum.ReadCommitted);
+
+            _asyncParameterValue = (AsyncProcessingServiceParameterValue)asyncReturnValue.Obj;
+
+            //Sets MaxThreads in ThreadPool.
+            bool SetMaxThreadsResult = ThreadPool.SetMaxThreads(10, 10);
+
+            //Main thread assigned task to worker thread in threadpool
+            ThreadStart threadStart = new ThreadStart(() => StartByThreadPool(_asyncParameterValue));
+            _workerThread = new Thread(threadStart);
+            _workerThread.Start();
+
+            //Update in mainthread
+            _asyncParameterValue = new AsyncProcessingServiceParameterValue("AsyncProcessingService", "Update", "Update", "SQL",
+                                                                            new MyUserInfo("AsyncProcessingService", "AsyncProcessingService"));
+
+            DbEnum.IsolationLevelEnum iso = DbEnum.IsolationLevelEnum.DefaultTransaction;
+
+            asyncReturnValue = (AsyncProcessingServiceReturnValue)myBusiness.DoBusinessLogic((AsyncProcessingServiceParameterValue)_asyncParameterValue, iso);
```

Comment:
Updates the status to the processing before starting the asynchronous processing.